prov/rxm/src/rxm_ep.c

```diff
@@ -851,9 +851,8 @@ static ssize_t rxm_ep_recvmsg(struct fid_ep *ep_fid, const struct fi_msg *msg,
 					     util_ep.ep_fid.fid);
 
 	return rxm_ep_recv_common_flags(rxm_ep, msg->msg_iov, msg->desc, msg->iov_count,
-					msg->addr, 0, 0, msg->context,
-					flags, (rxm_ep_rx_flags(rxm_ep) & FI_COMPLETION),
-					&rxm_ep->recv_queue);
+					msg->addr, 0, 0, msg->context, flags,
+					rxm_ep->util_ep.rx_msg_flags, &rxm_ep->recv_queue);
```

Comment:
Why does the code take 2 flags args?

Reply:
The flags can be ORed together. The code needs to be refactored.